### PR TITLE
Reference position traverser optimization/refactor

### PR DESCRIPTION
**Pull request by jorenham**
Runs in ~1s for the 100 strains data set.

Introduced `BreadthFirstIterator` which iterates the graph using "true" breadth-first search.

Will be merged at _23:59 03-06-2015_

### CODE REVIEW

File: src/main/java/nl/tudelft/lifetiles/graph/model/BreadthFirstIterator.java [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+    }
+
+    /**
+     * Check if the iterator has to wait for a vertex.
+     *
+     * @param vertex
+     *            the vertex, aka "it"
+     * @return whether to wait
+     */
+    private boolean waitForIt(final V vertex) {
```
Comment (albert-smit) [inline src/main/java/nl/tudelft/lifetiles/graph/model/BreadthFirstIterator.java]: isn't waitForVertex a better name?

**Comment by albert-smit** [comment]
looks good

**Comment by joswinter** [comment]
Looks good and works fast.

**Comment by alangerak** [comment]
Looks good

**Pull request merged**